src/server/game/Entities/Player/Player.cpp

```diff
@@ -1887,9 +1887,25 @@ void Player::Regenerate(Powers power)
                 }
             }
             break;
-        case POWER_ENERGY:                                  // Regenerate energy (rogue)
-            addvalue += 0.01f * m_regenTimer * sWorld->getRate(RATE_POWER_ENERGY);
+        case POWER_ENERGY: // Regenerate energy (rogue)
+        {
+            float baseRegenRate = 1.0f; // 1 energy per 0.1 seconds
+            float hasteMultiplier = 1.0f; // Start with no haste
+
+            // Apply haste effects from rogue talents
+            if (HasAuraEffect(13750, 0, ObjectGuid())) // Adrenaline Rush
+                hasteMultiplier *= 2.0f; // 100% faster regeneration
+
+            if (HasAuraEffect(58426, 0, ObjectGuid())) // Overkill
+                hasteMultiplier *= 1.3f; // 30% faster regeneration
+
+            if (HasAuraEffect(61329, 0, ObjectGuid())) // Vitality
+                hasteMultiplier *= 1.25f; // 25% faster regeneration
```

Comment:
I believe the logic is incorrect. Theoretically, if the player has all the specified auras, they would receive a 325% boost, but I think we’re expecting a 255% boost, right?